src/SparqlHandler.js

```diff
@@ -33,18 +33,19 @@ export default class SparqlHandler {
       throw new Error(`${pathData} should at least contain a subject and a predicate`);
 
     // Create triple patterns
-    let queryVar = '?subject', sorts = [], clauses = [];
+    let queryVar = '?subject', sorts = [], clauses = [], filters = [];
     if (pathExpression.length > 1) {
       queryVar = this.createVar(pathData.property);
-      ({ queryVar, sorts, clauses } = this.expressionToTriplePatterns(pathExpression, queryVar));
+      ({ queryVar, sorts, clauses, filters } = this.expressionToTriplePatterns(pathExpression, queryVar));
     }
     if (pathData.finalClause)
       clauses.push(pathData.finalClause(queryVar));
 
     // Create SPARQL query body
     const distinct = pathData.distinct ? 'DISTINCT ' : '';
     const select = `SELECT ${distinct}${pathData.select ? pathData.select : queryVar}`;
-    const where = ` WHERE {\n  ${clauses.join('\n  ')}\n}`;
+    const filter = filters.length ? `FILTER(${filters.join(' || ')})` : '';
```

Comment:
Just making a note for future that we will probably want to extend this to enable intersections at some stage